src/services/completions.ts

```diff
@@ -2049,6 +2050,7 @@ export function getCompletionEntriesFromSymbols(
 ): UniqueNameSet {
     const start = timestamp();
     const variableDeclaration = getVariableDeclaration(location);
+    const parameterDeclaration = getParameterDeclaration(contextToken);

```

Comment:
So your decision is to unify `variableDeclaration` and `parameterDeclaration`?

eg

```ts
const variableOrParameterDeclaration = getVariableOrParameterDeclaration(location, contextToken)
```

But what about this case then?:

```ts
const foo = (a = foo/*no foo suggestion*/) => {
    console.log(foo)
}
```

Reply:
BTW its fully valid location for `foo` suggestion, so maybe you're right

Reply:
I switched to `getVariableOrParameterDeclaration(contextToken: Node | undefined)`. I'm using only `contextToken` to fit logic in single `findAncestor`. I don't really the idea of dropping `location` as I feel it can introduce a lot of breakage, though I'll look at tests and try to fix them

Reply:
> BTW its fully valid location for foo suggestion, so maybe you're right

FYI https://github.com/microsoft/TypeScript/blob/0791bb0f8405058877c9930d63457f32522ae27c/tests/cases/fourslash/completionListWithoutVariableinitializer.ts#L50